src/Device/sdfuOperations.ts

```diff
@@ -32,18 +36,22 @@ export type PromiseChoice = (
 ) => Promise<string>;
 
 let lastMSG = '';
-const progressJson = ({
-    progressJson: progress,
-}: nrfDeviceLib.Progress.CallbackParameters) => {
-    const message = progress.message || '';
-
-    const status = `${message.replace('.', ':')} ${
-        progress.progressPercentage
-    }%`;
-
-    if (status !== lastMSG) logger.info(status);
-    lastMSG = status;
-};
+const progressJson =
+    ({ progressJson: progress }: nrfDeviceLib.Progress.CallbackParameters) =>
+    (dispatch: TDispatch) => {
+        const message = progress.message || '';
+
+        const status = `${message.replace('.', ':')} ${
+            progress.progressPercentage
+        }%`;
+
+        if (status !== lastMSG) {
+            dispatch(setDeviceSetupProgress(progress.progressPercentage));
+            dispatch(setDeviceSetupProgressMessage(status));
+            logger.info(status);
+            lastMSG = status;
+        }
+    };
```

Comment:
Would like to see this extracted for reuse since it's the same as in jprogOperations.ts